File: domain/src/main/java/hulkdx/com/domain/repository/MedicineRepository.java
package hulkdx.com.domain.repository;

import hulkdx.com.domain.models.MedicineCollection;
import io.reactivex.Single;

/**
 * Created by Mohammad Jafarzadeh Rezvan on 10/11/2018.
 */
public interface MedicineRepository {
    Single<MedicineCollection> getMedicine();
}

File: domain/src/main/java/hulkdx/com/domain/repository/MedicineRepository.kt
package hulkdx.com.domain.repository

import hulkdx.com.domain.models.MedicineCollection
import io.reactivex.Single

/**
 * Created by Mohammad Jafarzadeh Rezvan on 10/11/2018.
 */
interface MedicineRepository {
    fun getMedicines():                   Single<MedicineCollection>
    fun findMedicine(searchText: String): Single<MedicineCollection>
}
